### ISSUE TITLE
Incorrect rendering of the Annotation icon in Image captions

### ISSUE THREAD

**Issue by MrEddX** [squidfunk/mkdocs-material#7362]
### Context

Trying to insert annotation in an image caption.

### Bug description

Adding an image with annotation in the image caption, results in incorrect _half-rendered_ icon.

| Not clicked icon | Clicked icon |
|---|---|
| ![image](https://github.com/user-attachments/assets/9611b040-ab91-4b97-a8f8-1051021db760) | ![image](https://github.com/user-attachments/assets/a5d5068c-0933-45b6-9f6a-f21cd22325ac) |

### Related links

Already discussed in #7357

### Reproduction

[9.5.28-annotation-in-image-caption.zip](https://github.com/user-attachments/files/16203065/9.5.28-annotation-in-image-caption.zip)

### Steps to reproduce

1. Insert image with image caption plus annotation;
<details>

<summary>Details</summary>

```
...
<figure markdown>
  ![Image title](https://dummyimage.com/600x400/){ width=300 }
  <figcaption markdown>Image caption (1) text
  { .annotate }

  1. And the annotation text.

  </figcaption>
</figure>
...
```

</details>

2. Run `mkocs serve`
3. Open the page with the inserted image.

### Browser

Edge, Firefox

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thank for reporting and providing the reproduction! Fixed in 33452c92eaa8f6b9d0ea0cdde08c7539d559db1c. This commit also resets the font style back to the initial value (= not italic), which was also erroneously applied.

<img width="626" alt="Bildschirm­foto 2024-07-13 um 22 53 32" src="https://github.com/user-attachments/assets/1d61d412-c6ff-483b-a701-1963a629f17f">

**Comment by squidfunk** [comment]
Released as part of [9.5.29](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.29).
